pkgs/build-support/node/fetch-npm-deps/src/main.rs

```diff
@@ -71,9 +90,14 @@ fn to_new_packages(
             format!("{name}-{}", package.version),
             Package {
                 resolved: if let Ok(url) = Url::parse(&package.version) {
-                    Some(url)
+                    Some(url.into())
                 } else {
-                    package.resolved.as_deref().map(Url::parse).transpose()?
+                    package
+                        .resolved
+                        .as_deref()
+                        .map(Url::parse)
+                        .transpose()?
+                        .map(UrlOrString::from)
```

Comment:
Nit: I'm not sure, but wonder if the code can read clearer using the `UrlOrString` trick for `version` and `resolved`.